ui/push-health/PlatformConfig.jsx

```diff
@@ -22,12 +22,28 @@ class PlatformConfig extends React.PureComponent {
     };
   }
 
+  componentDidMount() {
+    const {
+      expandedDefaultMetric,
+      failure: { jobs },
+    } = this.props;
+    if (expandedDefaultMetric) {
+      const filteredJobs = jobs.filter(
+        (job) => job.id === parseInt(expandedDefaultMetric, 10),
+      );
+      if (filteredJobs.length > 0) {
+        this.setState({ selectedTask: filteredJobs[0], detailsShowing: true });
+      }
+    }
+  }
+
   setSelectedTask = (task) => {
     const { selectedTask } = this.state;
 
     if (selectedTask === task || !task) {
       this.setState({ selectedTask: null, detailsShowing: false });
     } else {
+      this.props.updateParamsAndState({ expandedDefaultMetric: task.id });
```

Comment:
A `metric` is different than a `task`.  I think this `state` name should be `selectedTask`.  Or you could use `selectedTaskId` for the querystring param name.

`metric`would be either `tests`, `builds` or `linting`.  And we should persist that, too.

The url params are seeming a bit verbose to me.  I'm wondering about something closer to this:

`selectedTask` = `task.id`
`test`= `test.id` (rather than the full test text)
`testgroup` = one of `pr`(possible regressions), `ki` (known issues)
`metric` = one of `tests`, `linting`, or `builds`

It also seems worthwhile to add these settings to the URL as you expand each one.  You may not want to necessarily share just a selected task, but an expanded list of tests in one area.

Those names might seem a bit generic for being part of `state`.  Perhaps `selectedTest`, `selectedTestGroup`, etc would be fine.

Reply:
> metricwould be either tests, builds or linting. And we should persist that, too.

I think we can have that without persisting it on the URL. If any of the params, that have been added in this PR is present, tests should be shown by default. The cost of this would be checking max 7 conditions on the React code

> test= test.id (rather than the full test text)

The tests are actually grouped by path or platform. So under any tests object in the frontend(which can be expanded or collapse), there are several tests from different failure lines. 
![Screenshot from 2020-08-05 21-06-08](https://user-images.githubusercontent.com/44131389/89436319-b2d10b80-d763-11ea-87c6-593a88023d1d.png)
This is the screenshot of the props in the Test Component. the ```test``` object may have more than one tests from different failure lines, hence different   ```failure.id```. This was my understanding from the code. Please rectify me if I am wrong or we have store the id which has the whole test name in the URL

Reply:
> This is the screenshot of the props in the Test Component. the `test` object may have more than one tests from different failure lines, hence different `failure.id`. This was my understanding from the code. Please rectify me if I am wrong or we have to store the id which has the whole test name in the URL

Ahh yes.  I see what you mean.  That ID won't be consistent, especially if some of the back-end data changes and a task is moved to "known issues" because retriggers come back with more passing than failing.

OK, let's go with the text, then.  I'll test this locally a bit today and see if I can find any issues.  But this is looking good.  :)